client/data/help/use-help-search-query.ts

```diff
@@ -14,41 +14,77 @@ export interface SearchResult {
 	source?: string;
 }
 
+interface TailoredArticles {
+	post_ids: Array< number >;
+	blog_id: number;
+	locale: string;
+}
+
 interface APIFetchOptions {
 	global: boolean;
 	path: string;
 }
 
-const fetchArticlesAPI = ( search: string, locale: string, sectionName: string ) => {
-	const queryString = buildQueryString( {
-		query: search,
-		locale,
-		section: sectionName,
+const fetchArticlesFromWPCom = async ( queryString: string ): Promise< SearchResult[] > => {
+	return await wpcomRequest( {
+		path: `help/search/wpcom?${ queryString }`,
+		apiNamespace: 'wpcom/v2/',
+		apiVersion: '2',
 	} );
+};
 
-	if ( canAccessWpcomApis() ) {
-		return wpcomRequest( {
-			path: `help/search/wpcom?${ queryString }`,
-			apiNamespace: 'wpcom/v2/',
-			apiVersion: '2',
-		} );
-	}
-
-	return apiFetch( {
+const fetchArticlesFromAPI = async ( queryString: string ): Promise< SearchResult[] > => {
+	return await apiFetch( {
 		global: true,
 		path: `/help-center/search?${ queryString }`,
 	} as APIFetchOptions );
 };
 
+const fetchArticlesFromWP = async ( articles: TailoredArticles ): Promise< SearchResult[] > => {
+	const { post_ids, blog_id, locale } = articles;
+	return await wpcomRequest( {
+		path: `help/articles`,
+		apiNamespace: 'wpcom/v2/',
+		apiVersion: '2',
+		method: 'PUT',
+		body: { post_ids, blog_id, locale },
+	} );
+};
+
+const fetchArticlesAPI = async (
+	search: string,
+	locale: string,
+	sectionName: string,
+	articles: TailoredArticles | undefined
+): Promise< SearchResult[] > => {
+	const queryString = buildQueryString( { query: search, locale, section: sectionName } );
+	let articlesResponse: SearchResult[] = [];
+	let searchResultResponse: SearchResult[] = [];
+
+	if ( articles ) {
+		articlesResponse = await fetchArticlesFromWP( articles );
+	}
+
```

Comment:
If we already have articles from `fetchArticlesFromWP`, we don't need to do another search after it 👍